### ISSUE TITLE
TensorRT库集成计划

### ISSUE THREAD

**Issue by luotao1** [PaddlePaddle/Paddle#9572]
计划：实现一个全部由tensor-rt库运行的ResNet网络（即整个网络是一个大的TensorRTOp）
时间：预计30工作日/2人 
related: #8492 
1. 安装tensor-rt库
2. 写一个tensorrt_test.cc单测，了解tensorrt最基本的流程，可参考[tensorflow/contrib/tensorrt/tensorrt_test.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensorrt/tensorrt_test.cc)
3. 新增TensorRTOp，可参考：[caffe2/contrib/tensorrt/tensorrt_op_trt.cc](https://github.com/yinghai/caffe2/blob/43b687793b3e4756d08eb4fa0313302bee2a2ed6/caffe2/contrib/tensorrt/tensorrt_op_trt.cc) 和 [tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensorrt/kernels/trt_engine_op.cc)
4. convert一系列op（即输入输出tensor的转变，和调用tensor-rt的api），可参考：[register_op_converters](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensorrt/convert/convert_nodes.cc#L2045) 和 [ConvertPool](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensorrt/convert/convert_nodes.cc#L1213)，每个convert都需要单测验证正确性。
5. merge batch norm：tensorflow里有[ConvertFusedBatchNorm](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensorrt/convert/convert_nodes.cc#L1834)，如果我们已经做了merge batch norm，就不需要额外转了。
6. 从fluid到tensor-rt的transpiler：从fluid的program_desc中删除原有的op，添加TensorRTOp。可参考：[tensorflow/contrib/tensorrt/convert/convert_graph.cc](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensorrt/convert/convert_graph.cc)
7. 对可以全部由tensor-rt op组成的网络，验证正确性，测试性能，可添加完整网络的单测。

![tensorrt](https://user-images.githubusercontent.com/6836917/38190703-038632fe-3698-11e8-8166-2104ad5615f9.PNG)

**Comment by shanyi15** [comment]
您好，此issue在近一个月内暂无更新，我们将于今天内关闭。若在关闭后您仍需跟进提问，可重新开启此问题，我们将在24小时内回复您。因关闭带来的不便我们深表歉意，请您谅解~感谢您对PaddlePaddle的支持! 
 Hello, this issue has not been updated in the past month. We will close it today for the sake of other user‘s experience. If you still need to follow up on this question after closing, please feel free to reopen it. In that case, we will get back to you within 24 hours. We apologize for the inconvenience caused by the closure and thank you so much for your support of PaddlePaddle Group!
